source/lac/petsc_matrix_base.cc

```diff
@@ -489,6 +489,89 @@ namespace PETScWrappers
   }
 
 
+  namespace internals
+  {
+    void perform_mmult (const MatrixBase &inputleft,
+                        const MatrixBase &inputright,
+                        MatrixBase       &result,
+                        const VectorBase &V,
+                        const bool        transpose_left)
+    {
+      const bool use_vector = (V.size() == inputright.m() ? true : false);
+      if (transpose_left == false)
+        {
+          Assert (inputleft.n() == inputright.m(),
+                  ExcDimensionMismatch(inputleft.n(), inputright.m()));
+        }
+      else
+        {
+          Assert (inputleft.m() == inputright.m(),
+                  ExcDimensionMismatch(inputleft.m(), inputright.m()));
+        }
+
+      result.clear();
+
+      PetscErrorCode ierr;
+
+      if (use_vector == false)
+        {
+          if (transpose_left)
+            {
+              ierr = MatTransposeMatMult (inputleft,
+                                          inputright,
+                                          MAT_INITIAL_MATRIX,
+                                          PETSC_DEFAULT,
+                                          &result.petsc_matrix());
```

Comment:
I guess this is the place where you are using the conversion function. Do you get an error if you just write `result.matrix`?

Reply:
Yes. If I just write `&result.matrix` I got
> petsc_matrix_base.h:960:9: error: ‘_p_Mat* dealii::PETScWrappers::MatrixBase::matrix' is protected within this context

I think this subtle issue is because we are using `Mat*` instead of `Mat` in this function.